### ISSUE TITLE
Face Mask Detection using  OpenCV

### ISSUE THREAD

**Issue by prathimacode-hub** [prathimacode-hub/ML-ProjectKart#40]
Hey LGM-SoC participants, good to see you.

The project helps in detecting whether the person is wearing face mask or not.

Dataset : https://www.kaggle.com/omkargurav/face-mask-dataset

If you like to work on this issue, follow the given guidelines for code as well as for the README.

Enjoy your open source journey!😎

**Comment by Chanakya-Sharma** [comment]
Can you assign me this

**Comment by prathimacode-hub** [comment]
Assigning issues would start from tomorrow, 10AM on first come first serve basis. So stay tuned. You will get all the details on how to get it assigned by today evening in slack channel. @Chanakya-Sharma

**Comment by Soumayan-pal01** [comment]
Hey @prathimacode-hub I would really like to work on this issue using the Haar Cascades and Caffe framework approaches as a LGM SOC participant. Please assign me this issue if possible. Thank you.

**Comment by prathimacode-hub** [comment]
Glad to know. Kindly express your interest tomorrow on 10th June, starting from 10AM. Issues would assigned on first come first serve basis. So stay tuned and act promptly after stipulated time. @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
> Glad to know. Kindly express your interest tomorrow on 10th June, starting from 10AM. Issues would assigned on first come first serve basis. So stay tuned and act promptly after stipulated time. @Soumayan-pal01

Okay, Thank you for the much valued information.

**Comment by Soumayan-pal01** [comment]
Hey @prathimacode-hub  I would really like to work on 'Model Creation' for this issue as a LGM-SOC participant, using the Haar Cascades and Caffe framework approaches for face detection and a finetuned MobileNetV2 model to detect masks on face taking real time video stream as input. Please assign me this issue if possible. Thank you.

**Comment by prathimacode-hub** [comment]
Hey, @Soumayan-pal01 you can work on onle 1 issue at a time. Kindly choose the issue you're interested to work on first.

**Comment by Soumayan-pal01** [comment]
> Hey, @Soumayan-pal01 you can work on only 1 issue at a time. Kindly choose the issue you're interested to work on first.

Okay I would like to work on this issue first . But can I work on issues where only I have commented alone like in issue #13 along with this one as the main purpose is to give everyone a fair chance? Please clarify, thank you.

**Comment by prathimacode-hub** [comment]
It's not right to keep other issue reserved while you do a task. If you want to work on this issue. I shall assign you this task. Also when you expres interest, also mention about task you will be dealing with. So it eases others with options. Model creation or front end or back end or designing. @Soumayan-pal01

**Comment by prathimacode-hub** [comment]
Loan eligibility prediction has been assigned to you already @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
> Loan eligibility prediction has been assigned to you already @Soumayan-pal01

Then please reassign me this issue as I would like to work on this issue first and I would like to work on the Model creation part. Thank you.

**Comment by prathimacode-hub** [comment]
Issue assigned.

**Comment by Soumayan-pal01** [comment]
Thank you for assigning this issue. There are a couple of doubts actually am having. Firstly, how to share my progress in every 2 days and the other one is what to put in the requirements.txt file. Can you please clarify @prathimacode-hub

**Comment by prathimacode-hub** [comment]
Sharing progress is letting us know of how much you had worked on and what all you could do till date. requirements.txt is a text file where you add dependent libraries to let users know that the project wouldn't work without it. @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
Thank you for the information @prathimacode-hub . Once I have completed working on this issue and pushed it in my forked repo I will inform here.

**Comment by prathimacode-hub** [comment]
Sure, let me know of it @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
Hey @prathimacode-hub since I have created this project taking real time video stream as input, therefore the efficiency of both the approaches can be tested only through video stream. Unlike the training set accuracy which is visible in the notebook, the testing models cannot show such accuracy as they real time inputs and deliver the predictions . So, should I upload demonstration videos for both the approaches for more clarity?

**Comment by prathimacode-hub** [comment]
Yeah, you should. Share the work samples before you create a PR. Let me see how would that look like. @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
Hey @prathimacode-hub I have completed working on this issue. I am providing the demonstration video link as well as my forked repo link. Please review if everything is properly done or not.

Video link - https://github.com/Soumayan-pal01/ML-ProjectKart/blob/main/Face%20Mask%20Detection%20using%20OpenCV/Images/Demonstration%20video%20of%20both%20models(Compressed).mp4

forked repo link - https://github.com/Soumayan-pal01/ML-ProjectKart/tree/main/Face%20Mask%20Detection%20using%20OpenCV

I have also updated some snapshots also in the README.md file in the 'Model' folder. Please inform can I create a PR or not. Thank you.

**Comment by prathimacode-hub** [comment]
Perfect. You had done an amazing work. Kindly proceed to raise your PR. @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
Is there any specific format to raise a PR or just give a overview of the project @prathimacode-hub

**Comment by prathimacode-hub** [comment]
Yeah, there is a PR template to follow from README file of the project. @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
Please review my PR @prathimacode-hub . Thank you.
